framework/db/ActiveQuery.php

```diff
@@ -486,6 +486,16 @@ private function buildJoinWith()
         }
         $this->join = array_values($uniqueJoins);
 
+        // https://github.com/yiisoft/yii2/issues/16092
+        $uniqueJoinsByTableName = [];
```

Comment:
Does this work for aliased tables or only table names?
It should work primarily on aliases, and only use table name if an alias is not present.

Reply:
It works for both aliased tables and table names.